drivers/include/si7021.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2016 Bas Stottelaar <basstottelaar@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    driver_si7021 SI7021
+ * @ingroup     drivers
+ * @brief       Driver for the SI7021 temperature and humidity sensor.
+ * @{
+ *
+ * @file
+ * @brief       Interface definition of the SI7021 driver.
+ *
+ * @author      Bas Stottelaar <basstottelaar@gmail.com>
+ */
+
+#ifndef SI7021_H_
+#define SI7021_H_
+
+#include <stdint.h>
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name SI7021 address
+ */
+#define SI7021_ADDRESS              (0x80)
+
+/**
+ * @name SI7021 device commands
+ * @{
+ */
+#define SI7021_MEASURE_RH_HOLD      (0xE5)
+#define SI7021_MEASURE_RH           (0xF5)
+#define SI7021_MEASURE_TEMP_HOLD    (0xE3)
+#define SI7021_MEASURE_TEMP         (0xF3)
+#define SI7021_MEASURE_TEMP_PREV    (0xE0)
+#define SI7021_RESET                (0xFE)
+#define SI7021_WRITE_USER_REG       (0xE6)
+#define SI7021_READ_USER_REG        (0xE7)
+#define SI7021_WRITE_HEATER_REG     (0x51)
+#define SI7021_READ_HEATER_REG      (0x11)
+#define SI7021_READ_ID_FIRST_A      (0xFA)
+#define SI7021_READ_ID_FIRST_B      (0x0F)
+#define SI7021_READ_ID_SECOND_A     (0xFC)
+#define SI7021_READ_ID_SECOND_B     (0xC9)
+#define SI7021_READ_REVISION_A      (0x84)
+#define SI7021_READ_REVISION_B      (0xB8)
+/** @} */
+
+/**
+ * @name SI7021 register values
+ * @{
+ */
+#define SI7021_ID                   (0x15)
+
+#define SI7021_REVISION_1           (0xFF)
+#define SI7021_REVISION_2           (0x20)
+/** @} */
+
+/**
+ * @brief SI7021 device descriptor
+ */
+typedef struct {
+    i2c_t i2c_dev;              /**< I2C bus the sensors is connected to */
+} si7021_t;
+
+/**
+ * @brief   Test if the device id and revision number are as expected.
+ *
+ * @param[in] dev           device descriptor
+ * @return                  zero on succesful test
+ * @return                  non-zero on unsuccesfull test.
+ */
+int si7021_test(si7021_t *dev);
+
+/**
+ * @brief   Read the revision number of the sensor firmware.
+ *
+ * @param[in] dev           device descriptor
+ * @return                  zero on succesful initialization.
+ * @return                  non-zero on error
+ */
+int si7021_init(si7021_t *dev, i2c_t i2c_dev);
+
+/**
+ * @brief   Reset the sensor. This takes at most 25 milliseconds.
+ *
+ * @param[in] dev           device descriptor
+ */
+void si7021_reset(si7021_t *dev);
+
+/**
+ * @brief   Read the relative humidity from the sensor. Uses clock streching.
+ *
+ * @param[in] dev           device descriptor
+ * @return                  relative humidity between (0 - 100%)
+ */
+float si7021_get_relative_humidity(si7021_t *dev);
```

Comment:
I don't really like the usage of `float` here, as this pulls in quite some sw-floating point code for many platforms. Fixed point is the better choice here, so for the humidity I would return hundrets of percent (`uint16_t` and the temperature I would return as thousands of degree (`int32_t`).